psycopg/psycopg/types/enum.py

```diff
@@ -0,0 +1,95 @@
+"""
+Adapters for the enum type.
+"""
+from enum import Enum
+from typing import Optional, TypeVar, Generic, Type
+
+from .string import StrBinaryDumper, StrDumper
+from .. import postgres
+from .._typeinfo import EnumInfo as EnumInfo  # exported here
+from ..abc import AdaptContext
+from ..adapt import Buffer, Loader
+from ..pq import Format
+
+E = TypeVar("E", bound=Enum)
+
+
+class EnumLoader(Loader, Generic[E]):
+    format = Format.TEXT
+    python_type: Type[E]
+
+    def __init__(self, oid: int, context: Optional[AdaptContext] = None):
+        super().__init__(oid, context)
+
+    def load(self, data: Buffer) -> E:
+        if isinstance(data, memoryview):
+            data = bytes(data)
+        return self.python_type(data.decode())
+
+
+class EnumBinaryLoader(EnumLoader[E]):
+    format = Format.BINARY
+
+
+class EnumDumper(StrDumper):
```

Comment:
Can I use `StrDumper` as base class, but set custom `oid` in attributes?

Reply:
That would only work for str enum and might break if the connection is sql_ascii (because it would return bytes, not str, for reasons explained [here](https://www.psycopg.org/psycopg3/docs/basic/adapt.html#strings-adaptation)).